### PR TITLE
Add Progress Bar Block

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
 		"@semantic-release/changelog": "^5.0.1",
 		"@semantic-release/exec": "^5.0.0",
 		"@semantic-release/git": "^9.0.0",
+		"@wordpress/block-editor": "^4.3.7",
```
Comment (HardeepAsrani) [inline package.json:38]: Why did you add that?

File: src/blocks/progress-bar/attributes.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const attributes = {
+	height: {
+		type: 'number',
+		default: 35
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/attributes.js:4]: Don't need a default. The default can be from the CSS.

File: src/blocks/progress-bar/attributes.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		type: 'number',
+		default: 50
+	},
+	duration: {
+		type: 'number',
+		default: 3
+	},
+	title: {
+		type: 'string',
+		default: 'Skill'
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/attributes.js:16]: Use `__( 'Skills' )` to make it translatable.

File: src/blocks/progress-bar/constants.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export const fontRatio = 0.342;
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/constants.js:1]: No need for it to be a separate file for a single constant.

File: src/blocks/progress-bar/edit.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+				heightMode={ heightMode }
+				setHeightMode={ setHeightMode }
+			/>
+			<div className="wp-themeisle-progress-bar-block">
+
+				{
+					( 'outer' === attributes.titleStyle || 'outer' === attributes.percentagePosition ) && (
+						<div className="wp-themeisle-progress-bar-outer-content">
+							{'outer' === attributes.titleStyle && (
+								<h3 className="wp-themeisle-block-progress-bar-outer-content__title">
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/edit.js:83]: Don't use heading here. Span is fine.

File: src/blocks/progress-bar/edit.js [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+									{`${ attributes.title }`}
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/edit.js:84]: Why using backticks?

File: src/blocks/progress-bar/edit.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+								</h3>
+							)}
+							{ 'outer' === attributes.percentagePosition && showPercentage && (
+								<div
+									className={
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/edit.js:89]: Don't need classnames when there are no conditions.

File: src/blocks/progress-bar/inspector.js [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+	return (
+		<InspectorControls>
+			<Fragment>
+				<PanelBody
+					title={ __( 'Settings' ) }
+					initialOpen={ true }
+				>
+					<TextControl
+						label="Title"
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/inspector.js:76]: Make labels translatable.

File: src/blocks/progress-bar/inspector.js [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+						onChange={ onDurationChange }
+						min={ 0 }
+						max={ 3 }
+						step={ 0.1 }
+					/>
+
+					{
+						( 35 <= attributes.height ) && (
+							<SelectControl
+								label="Title Style"
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/inspector.js:103]: Make labels translatable.

File: src/blocks/progress-bar/inspector.js [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+								value={ attributes.titleStyle }
+								options={ [
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/inspector.js:105]: Make labels translatable.

File: src/blocks/progress-bar/inspector.js [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+									{ label: 'Simple', value: 'simple' },
+									{ label: 'Outer', value: 'outer' }
+								] }
+								onChange={ selectTitleStyle }
+							/>
+						)
+					}
+
+					<SelectControl
+						label="Show Percentage"
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/inspector.js:116]: Make labels translatable.

File: src/blocks/progress-bar/inspector.js [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+						value={ attributes.percentagePosition }
+						options={ [
```
Comment (HardeepAsrani) [inline src/blocks/progress-bar/inspector.js:118]: Make labels translatable.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
You can make these changes and submit them. I'll make few changes with the Save function which includes bringing CSS out of this once you're done.

File: package.json (hunk 1 above)
Comment (Soare-Robert-Daniel) [reply package.json:38]: this was for the QR Code, didn't pay attention and I installed in the wrong place.

File: src/blocks/progress-bar/attributes.js (hunk 2 above)
Comment (Soare-Robert-Daniel) [reply src/blocks/progress-bar/attributes.js:4]: I can't do that, ResizableBox depends on it.

Comment (HardeepAsrani) [reply src/blocks/progress-bar/attributes.js:4]: Got it.

**Review by HardeepAsrani — APPROVED** [review]
